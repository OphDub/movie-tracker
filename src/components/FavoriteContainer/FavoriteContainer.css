.FavoriteContainer {
  display: flex;
  flex-direction: row;
  background: rgba(100,100,255, 0.1);
  -webkit-perspective: 1000;
  -moz-perspective: 1000;
  -o-perspective: 1000;
  perspective: 1000;
}