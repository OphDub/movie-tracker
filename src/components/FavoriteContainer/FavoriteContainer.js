import React, { Component } from 'react';
import Card from '../Card/Card';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { getAllFavorites, deleteFavorite, addFavorite } from '../../api';
import { addFavoriteAction } from '../../actions/actionIndex';

export class FavoriteContainer extends Component {
  
  handleFavorite = async (movie) => {
    this.props.activeUser ? this.toggleFavorite(movie) : this.props.history.push('login')
  }

  toggleFavorite = async (movie) => {
    const user_id = this.props.activeUser.id
    const allFavs = await getAllFavorites(user_id);
    const match = allFavs.data.filter(favMovie => favMovie.movie_id === movie.movie_id)
    
    movie.user_id = user_id;

    if (match.length > 0) {
      const remaining = allFavs.data.filter(favMovie => favMovie.movie_id !== movie.movie_id)
      deleteFavorite(user_id, movie.movie_id ) 
      this.props.addFavorite(remaining)
    } else {
      addFavorite(movie)
    }
  }

  getFavorites = async () => {
    const allFavs = await getAllFavorites(this.props.activeUser.id);
    this.props.addFavorite(allFavs.data);
  }

  componentDidMount () {
    // this.getFavorites()
  }

  cardsArray () {
    return this.props.favoriteArray.map(movie => 
      <Card 
        movie={movie}
        id={movie.movie_id}
        key={movie.movie_id}
        handleFavorite={this.handleFavorite}
      />
    )
  }
  
  render () {
    return (
      <div>
        {this.cardsArray()}
      </div>
    )
  }
}

const mapState = (state) => ({
  activeUser: state.activeUser,
  favoriteArray: state.favoriteArray,
})

const mapDispatch = (dispatch) => ({
  addFavorite: (favoriteData) => dispatch(addFavoriteAction(favoriteData)),
})

export default withRouter(connect(mapState, mapDispatch)(FavoriteContainer))