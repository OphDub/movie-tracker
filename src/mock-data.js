 export const cleanMovieArray = [
    {
      movie_id: 399055,
      title: 'The Shape of Water',
      vote_average: 7.4,
      vote_count: 794,
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.',
      release_date: '2017-12-08',
      poster_path: 'https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg'
    },
    {
      "movie_id": 284054,
      "title": "Black Panther",
      "vote_average": 5,
      "vote_count": 72,
      "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
      "release_date": "2018-02-13",
      "poster_path": "https://image.tmdb.org/t/p/w500//bLBUCtMQGJclH36clliPLmljMys.jpg"
    }
] 

export const dirtyMovieObject = { results: [
    { 
      vote_average: 7.4,
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.',
      vote_count: 794,
      title: 'The Shape of Water',
      poster_path: '/iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg',
      release_date: '2017-12-08',
      id: 399055,
      randomnumber: 24341324142134,
    },
    {
      "title": "Black Panther",
      "vote_count": 72,
      "thing": "asdfasdfasdf",
      "id": 284054,
      "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
      "release_date": "2018-02-13",
      "poster_path": "/bLBUCtMQGJclH36clliPLmljMys.jpg",
      "vote_average": 5,
    }
]}

export const oneMovie = {
  movie_id: 284054,
  title: 'Coco',
  vote_average: 7.7,
  vote_count: 1987,
  overview: 'Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel\'s family history.',
  posterImage: 'https://image.tmdb.org/t/p/w500//eKi8dIrr8voobbaGzDpe8w0PVbC.jpg'
}

export const userData = {
  id: 2,
  name: 'brophus park',
  email: 'bros@gmail.com',
  password: 'pass',
}

export const movieRawData = {
  data: {
      movie_id: 399055,
      title: 'The Shape of Water',
      vote_average: 7.4,
      vote_count: 794,
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.',
      release_date: '2017-12-08',
      poster_path: 'https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg'
    }
}
