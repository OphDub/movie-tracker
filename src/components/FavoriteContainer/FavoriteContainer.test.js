import React from 'react';
import { FavoriteContainer } from './FavoriteContainer';
import { shallow } from 'enzyme';
import { cleanMovieArray, userData } from '../../mock-data.js';

describe('FavoriteContainer', () => {
  let wrapper;
  const mockFunction = jest.fn();
  
  beforeEach( () => {
    wrapper = shallow(
      <FavoriteContainer 
        favoriteArray={cleanMovieArray}
        activeUser={userData}
      />);
  });

  it('should match the snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });
});