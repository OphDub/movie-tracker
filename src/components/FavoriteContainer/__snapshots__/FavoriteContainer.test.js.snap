// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteContainer should match the snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoriteContainer
    activeUser={
      Object {
        "email": "bros@gmail.com",
        "id": 2,
        "name": "brophus park",
        "password": "pass",
      }
    }
    favoriteArray={
      Array [
        Object {
          "overview": "Despite his family’s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel's family history.",
          "posterImage": "https://image.tmdb.org/t/p/w500//eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
          "title": "Coco",
          "vote_average": 7.7,
          "vote_count": 1987,
        },
        Object {
          "overview": "Thomas leads his group of escaped Gladers on their final and most dangerous mission yet. To save their friends, they must break into the legendary Last City, a WCKD-controlled labyrinth that may turn out to be the deadliest maze of all. Anyone who makes it out alive will get answers to the questions the Gladers have been asking since they first arrived in the maze.",
          "posterImage": "https://image.tmdb.org/t/p/w500//7GgZ6DGezkh3szFdvskH5XD4V0t.jpg",
          "title": "Maze Runner: The Death Cure",
          "vote_average": 7.3,
          "vote_count": 326,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          handleFavorite={[Function]}
          id={undefined}
          movie={
            Object {
              "overview": "Despite his family’s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel's family history.",
              "posterImage": "https://image.tmdb.org/t/p/w500//eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
              "title": "Coco",
              "vote_average": 7.7,
              "vote_count": 1987,
            }
          }
        />,
        <Card
          handleFavorite={[Function]}
          id={undefined}
          movie={
            Object {
              "overview": "Thomas leads his group of escaped Gladers on their final and most dangerous mission yet. To save their friends, they must break into the legendary Last City, a WCKD-controlled labyrinth that may turn out to be the deadliest maze of all. Anyone who makes it out alive will get answers to the questions the Gladers have been asking since they first arrived in the maze.",
              "posterImage": "https://image.tmdb.org/t/p/w500//7GgZ6DGezkh3szFdvskH5XD4V0t.jpg",
              "title": "Maze Runner: The Death Cure",
              "vote_average": 7.3,
              "vote_count": 326,
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleFavorite": [Function],
          "id": undefined,
          "movie": Object {
            "overview": "Despite his family’s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel's family history.",
            "posterImage": "https://image.tmdb.org/t/p/w500//eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
            "title": "Coco",
            "vote_average": 7.7,
            "vote_count": 1987,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleFavorite": [Function],
          "id": undefined,
          "movie": Object {
            "overview": "Thomas leads his group of escaped Gladers on their final and most dangerous mission yet. To save their friends, they must break into the legendary Last City, a WCKD-controlled labyrinth that may turn out to be the deadliest maze of all. Anyone who makes it out alive will get answers to the questions the Gladers have been asking since they first arrived in the maze.",
            "posterImage": "https://image.tmdb.org/t/p/w500//7GgZ6DGezkh3szFdvskH5XD4V0t.jpg",
            "title": "Maze Runner: The Death Cure",
            "vote_average": 7.3,
            "vote_count": 326,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            handleFavorite={[Function]}
            id={undefined}
            movie={
              Object {
                "overview": "Despite his family’s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel's family history.",
                "posterImage": "https://image.tmdb.org/t/p/w500//eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
                "title": "Coco",
                "vote_average": 7.7,
                "vote_count": 1987,
              }
            }
          />,
          <Card
            handleFavorite={[Function]}
            id={undefined}
            movie={
              Object {
                "overview": "Thomas leads his group of escaped Gladers on their final and most dangerous mission yet. To save their friends, they must break into the legendary Last City, a WCKD-controlled labyrinth that may turn out to be the deadliest maze of all. Anyone who makes it out alive will get answers to the questions the Gladers have been asking since they first arrived in the maze.",
                "posterImage": "https://image.tmdb.org/t/p/w500//7GgZ6DGezkh3szFdvskH5XD4V0t.jpg",
                "title": "Maze Runner: The Death Cure",
                "vote_average": 7.3,
                "vote_count": 326,
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleFavorite": [Function],
            "id": undefined,
            "movie": Object {
              "overview": "Despite his family’s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel's family history.",
              "posterImage": "https://image.tmdb.org/t/p/w500//eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
              "title": "Coco",
              "vote_average": 7.7,
              "vote_count": 1987,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleFavorite": [Function],
            "id": undefined,
            "movie": Object {
              "overview": "Thomas leads his group of escaped Gladers on their final and most dangerous mission yet. To save their friends, they must break into the legendary Last City, a WCKD-controlled labyrinth that may turn out to be the deadliest maze of all. Anyone who makes it out alive will get answers to the questions the Gladers have been asking since they first arrived in the maze.",
              "posterImage": "https://image.tmdb.org/t/p/w500//7GgZ6DGezkh3szFdvskH5XD4V0t.jpg",
              "title": "Maze Runner: The Death Cure",
              "vote_average": 7.3,
              "vote_count": 326,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
